source/agora/node/Node.d

```diff
@@ -104,9 +106,25 @@ public class Node : API
     /// Enrollment manager
     protected EnrollmentManager enroll_man;
 
+    /// If a custom genesis block is set it will be stored here
+    private immutable Block genesis_block;
+
     /// Ctor
     public this (const Config config)
     {
+        // custom genesis block provided
+        if (config.node.genesis_block.length > 0)
+        {
+            import std.array;
+            import std.conv;
+
+            // hex => bin
+            auto block_bytes = config.node.genesis_block.chunks(2).map!(
+                twoDigits => twoDigits.parse!ubyte(16)).array();
+            this.genesis_block = block_bytes.deserializeFull!(immutable(Block));
+            setGenesisBlock(this.genesis_block);
```

Comment:
The block needs to be validated. What if the merkle root is invalid ?

Reply:
Should I add the validation here, or better yet into the `setGenesisBlock` routine?

Reply:
oh lol we don't have validation for the genesis block:

```
// special case for the genesis block
if (block.header.height == 0)
    return block == GenesisBlock ?
        null : "Block: Height 0 but not Genesis block";
```

That's bad.